web/server/config/server_config.json

```diff
@@ -54,11 +54,59 @@
         "adm", "cc-users"
       ]
     },
+    "method_oauth": {
+      "enabled": false,
+      "groups": [
+        "allowed_users"
+      ],
+      "providers" : {
+        "github" : {
+          "enabled" : false,
+          "oauth_client_id" : "ExampleClientID",
+          "oauth_client_secret": "exampleClientSecret",
+          "oauth_authorization_uri": "https://github.com/login/oauth/authorize",
+          "oauth_redirect_uri": "http://path_To_CodeChecker_Login_Page",
+          "oauth_token_uri": "https://github.com/login/oauth/access_token",
+          "oauth_user_info_uri": "https://api.github.com/user",
+          "oauth_scope": "openid email profile",
+          "oauth_user_info_mapping": {
+            "username": "login",
+            "email": "email",
+            "fullname": "name"
+          },
```

Comment:
I can't see a technical purpose of this mapping. Maybe this could be eliminated.

Reply:
the `email` and `full name` indeed had no purpose and were removed, but username is used as when you log in using `oauth` it Maps the username to login or email depending on needs and uses it to store info in database and to demonstrate name on site